controllers/dopplersecret_controller_secrets.go

```diff
@@ -244,13 +244,20 @@ func (r *DopplerSecretReconciler) UpdateSecret(ctx context.Context, dopplerSecre
 	if dopplerSecret.Spec.ManagedSecretRef.Namespace == "" {
 		dopplerSecret.Spec.ManagedSecretRef.Namespace = dopplerSecret.Namespace
 	}
+
+	// Handle namespace defaults
 	if dopplerSecret.Spec.TokenSecretRef.Namespace == "" {
 		dopplerSecret.Spec.TokenSecretRef.Namespace = dopplerSecret.Namespace
 	}
 
-	dopplerToken, err := r.GetDopplerToken(ctx, dopplerSecret)
+	authProvider, err := r.getAuthProvider(ctx, &dopplerSecret)
+	if err != nil {
+		return fmt.Errorf("Failed to get auth provider: %w", err)
+	}
+
+	apiContext, err := authProvider.GetAPIContext(ctx)
 	if err != nil {
-		return fmt.Errorf("Failed to load Doppler Token: %w", err)
+		return fmt.Errorf("Failed to get API context: %w", err)
```

Comment:
nit: If our error messages are also going to start with uppercase letters, let's refactor the newly added errors to be consistent there too.